src/libraries/System.IO.Hashing/src/System/IO/Hashing/XxHash32.State.cs

```diff
@@ -0,0 +1,113 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers.Binary;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+// Implemented from the specification at
+// https://github.com/Cyan4973/xxHash/blob/f9155bd4c57e2270a4ffbb176485e5d713de1c9b/doc/xxhash_spec.md
+
+namespace System.IO.Hashing
+{
+    public sealed partial class XxHash32
+    {
+        private struct State
+        {
+            private const uint Prime32_1 = 0x9E3779B1;
+            private const uint Prime32_2 = 0x85EBCA77;
+            private const uint Prime32_3 = 0xC2B2AE3D;
+            private const uint Prime32_4 = 0x27D4EB2F;
+            private const uint Prime32_5 = 0x165667B1;
+
+            private uint _acc1;
+            private uint _acc2;
+            private uint _acc3;
+            private uint _acc4;
+            private readonly uint _smallAcc;
+
+            internal State(uint seed)
+            {
+                unchecked
```

Comment:
Why are all of these `unchecked`s necessary?  I don't see CheckForOverflowUnderflow set in the csproj.

Reply:
VS squiggled a piece with an overflowed constant, so I just added a bunch.

Reply:
FWIW, I find it confusing when `unchecked` is used like this, as it suggests to a reader than anything not `unchecked` is checked, and that's not the case.  I'd prefer if it was only used in the places where it was necessary to suppress warnings for compile-time overflows.

Reply:
Looks like I was able to get rid of all but two of them (the high-bit constants in the tests, int 0xF00D_F00D and long 0xF00D_F00D_F00D_F00D).

Trying to replay the problem, it's  XxHash32+State ctor: `_acc1 = seed + Prime32_1 + Prime32_2`... except before I had the "seed +" in there.  Adding the two const ints overflows in constant folding, but since they're RHS they appear to be not folded.  Now that I've written that down it bothers me, so I added parentheses, so two uncheckeds came back.

Reply:
Thanks :)